src/Telemetry/AzureDevOpsTimeline/AzureDevOpsTimeline.cs

```diff
@@ -319,5 +364,56 @@ private static string GetBucket(AugmentedTimelineIssue augIssue)
         {
             return await azureServer.ListBuilds(project, cancellationToken, minDateTime, limit);
         }
+
+        private async Task GetImageNames(List<AugmentedTimelineRecord> records, CancellationToken cancellationToken)
+        {
+            SemaphoreSlim throttleSemaphore = new SemaphoreSlim(50);
```

Comment:
Please also check https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.parallel.foreach?view=net-6.0

Maybe it will get you what you want better?

```
Parallel.ForEach(records, new ParallelOptions {MaxDegreeOfParallelism = 10},
record =>
{
   // ...
});
```

Reply:
I think there was some problem with the Parallel.ForEach that @lpatalas had when we originally wrote this in the MatrixOfTruth project, could you remind me what it was please?